File: Src/java/engine/src/test/java/org/opencds/cqf/cql/engine/execution/EmptyStringsTest.java
package org.opencds.cqf.cql.engine.execution;

import static org.hamcrest.MatcherAssert.assertThat;
import static org.hamcrest.Matchers.is;
import static org.hamcrest.Matchers.nullValue;

import org.junit.jupiter.api.Test;

class EmptyStringsTest extends CqlTestBase {

    @Test
    void all_empty_string() {
        var results = engine.evaluate(toElmIdentifier("EmptyStringsTest"));
        var value = results.forExpression("Null").value();
        assertThat(value, is(nullValue()));

        value = results.forExpression("Space").value();
        assertThat(value, is(" "));

        value = results.forExpression("Empty").value();
        assertThat(value, is(""));
    }
}

File: Src/java/engine/src/test/kotlin/org/opencds/cqf/cql/engine/execution/EmptyStringsTest.kt
package org.opencds.cqf.cql.engine.execution

import org.hamcrest.MatcherAssert
import org.hamcrest.Matchers
import org.junit.jupiter.api.Test

internal class EmptyStringsTest : CqlTestBase() {
    @Test
    fun all_empty_string() {
        val results = engine.evaluate(toElmIdentifier("EmptyStringsTest"))
        var value = results.forExpression("Null")!!.value()
        MatcherAssert.assertThat(value, Matchers.`is`(Matchers.nullValue()))

        value = results.forExpression("Space")!!.value()
        MatcherAssert.assertThat(value, Matchers.`is`(" "))

        value = results.forExpression("Empty")!!.value()
        MatcherAssert.assertThat(value, Matchers.`is`(""))
    }
}
